hypervisor/src/arch/x86/emulator/instructions/mov.rs

```diff
@@ -27,27 +26,18 @@ macro_rules! mov_rm_r {
             state: &mut T,
             platform: &mut dyn PlatformEmulator<CpuState = T>,
         ) -> Result<(), EmulationError<Exception>> {
-            let src_reg_value = state
-                .read_reg(insn.op1_register())
+            let src_reg_value = get_op(&insn, 1, state, platform, std::mem::size_of::<$bound>())
                 .map_err(EmulationError::PlatformEmulationError)?;
 
-            match insn.op0_kind() {
-                OpKind::Register => state
-                    .write_reg(insn.op0_register(), src_reg_value)
-                    .map_err(EmulationError::PlatformEmulationError)?,
-
-                OpKind::Memory => {
-                    let addr = memory_operand_address(insn, state, true)
-                        .map_err(EmulationError::PlatformEmulationError)?;
-                    let src_reg_value_type: $bound = src_reg_value as $bound;
-
-                    platform
-                        .write_memory(addr, &src_reg_value_type.to_le_bytes())
-                        .map_err(EmulationError::PlatformEmulationError)?
-                }
-
-                k => return Err(EmulationError::InvalidOperand(anyhow!("{:?}", k))),
-            }
+            set_op(
+                &insn,
+                0,
+                state,
+                platform,
+                std::mem::size_of::<$bound>(),
```

Comment:
This is already present in the code and should not be fixed with this PR, but we should have more consistent macros arguments names, and switch to `bound` for all of them.